src/decoder.rs

```diff
@@ -196,11 +196,31 @@ impl<R: Read> Decoder<R> {
         Some(data)
     }
 
+    /// Heuristic to avoid starting thread, synchronization if we expect a small amount of
+    /// parallelism to be utilized.
+    fn select_worker(frame: &FrameInfo, worker_preference: PreferWorkerKind) -> PreferWorkerKind {
+        // TODO: figure out the exact value.
+        const PARALLELISM_THRESHOLD: u64 = 1024 * 1024;
```

Comment:
```suggestion
        const PARALLELISM_THRESHOLD: u64 = 128 * 128;
```

Extremely tall images with this area benefit from skipping multithreading (10% speedup) but extremely wide images with this area see no difference.